packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,301 @@
+    setContainerOpen(false);
+    if (inputRef.current) inputRef.current.focus();
+  };
+
+  // Handler for when an item is clicked
+  const handleSelect = item => {
+    setSelected(item);
+
+    setSearch(item[searchField]);
+    setContainerOpen(false);
+    if (inputRef.current) inputRef.current.focus();
+    if (onSelect) onSelect(item);
+  };
+
+  const handleScrolling = item => {
+    const selectedNode = [
+      ...document.querySelectorAll('div.typeahead-option'),
+    ].find(i => {
+      if (i.innerText && item.label) {
+        return i.innerText.toLowerCase() === item.label.toLowerCase();
+      }
+      return false;
+    });
+
+    const container = selectedNode?.parentElement?.parentElement;
+
+    if (!container || !selectedNode) return;
+
+    const containerHeight: number = container.getClientRects()[0].height;
+    const overScroll = selectedNode.offsetHeight / 3;
+
+    const scrollPos =
+      selectedNode.offsetTop +
+      selectedNode.clientHeight -
+      containerHeight +
+      overScroll;
+
+    container.scrollTop = scrollPos;
+  };
+
+  const handleKeyNavigation = (direction: number) => {
+    let selectedIndex;
+    let cursorIndex;
+    let newItem: OptionObject = options[0];
+    const optionsCount = options.length - 1;
+
+    // If there's an existing item, navigate from that position
+    if (selected) {
+      selectedIndex = options.findIndex(item => item.value === selected.value);
+      const newIndex = selectedIndex + direction;
+
+      // If we've reached the end, start at the top
+      if (newIndex > optionsCount) {
+        cursorIndex = 0;
+      }
+      // If we're at the top going backwards, start at the last item
+      else if (newIndex < 0) {
+        cursorIndex = optionsCount;
+      }
+      // Carry-on otherwise
+      else {
+        cursorIndex = newIndex;
+      }
+      newItem = options[cursorIndex];
+    }
+    setSelected(newItem);
+    setSearch(newItem[searchField]);
+
+    // Scrolling
+    handleScrolling(newItem);
+
+    if (onSelect) onSelect(newItem);
+  };
+
+  return (
+    <Box ref={forwardedRef}>
+      <TypeaheadInputField
+        label={label}
+        id={id}
+        value={search}
+        placeholder={placeholder}
+        size={size}
+        onChange={handleChange}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        onClear={handleClear}
+        onKeyNavigation={handleKeyNavigation}
+        setContainer={setContainerOpen}
+        ref={inputRef}
+      />
+
+      {containerOpen && inputRef.current && (
+        <Layer>
+          <Flyout
+            showCaret={false}
+            anchor={inputRef.current}
+            idealDirection="down"
+            onDismiss={() => {}}
+            positionRelativeToAnchor={false}
+            size="flexible"
+          >
+            <Box
+              position="relative"
+              padding={1}
+              marginTop={2}
+              marginBottom={2}
+              maxHeight="50vh"
```

Comment:
@simkessy @ashleyseto what was the resolution on the maxHeight? Should we use an absolute value instead of 50% of the viewport height? E.g. `400px`

Reply:
Last statement on this was that users shouldn't have control over the default. No specific number / height was provided.

Reply:
If we use an absolute value, it can go off the screen right? The goal is to keep it on the screen. The 50% seems to be working but I was curious how other design systems handle this component.